tests/diagnostic/include.h

```diff
@@ -14,12 +14,37 @@ using namespace PHARE::diagnostic::h5;
 constexpr unsigned NEW_HI5_FILE
     = HighFive::File::ReadWrite | HighFive::File::Create | HighFive::File::Truncate;
 
+template<typename HiFile, typename Field>
+void checkField(HiFile& file, Field& field, std::string path)
+{
+    std::vector<float> fieldV;
+    file.getDataSet(path).read(fieldV);
+    EXPECT_EQ(fieldV.size(), field.size());
+
+    for (size_t i = 0; i < fieldV.size(); i++)
+    {
+        if (!std::isnan(fieldV[i]))
+        {
+            EXPECT_FLOAT_EQ(fieldV[i], field.data()[i]);
```

Comment:
why not using field(i) instead of field.data()[i] ?

Reply:
ah because it's flattened in the file?

Reply:
.data() is used to write the h5 file, so I used it to verify also